theurgy.lic

```diff
@@ -55,13 +55,129 @@ def buy_wine
   put_in_container('wine')
 end
 
+def buy_Taffelberries
+  DRCI.order_item(8280, 14)
+  fput 'get Taffelberries'
+  put_in_container('Taffelberries')
+  fput 'get Taffelberries'
+  put_in_container('Taffelberries')
+end
+
 def clean_altar
   get_from_container($settings.water_holder)
   fput 'clean altar with water'
   waitfor 'You feel that your gods have smiled upon you for your attempts to please them'
   put_in_container($settings.water_holder)
 end
 
+def prayer_bead
+  return if DRSkill.getrank('Mechanical Lore') < 140
+
+  fput "stow #{GameObj.right_hand.noun}" if GameObj.right_hand.name != 'Empty'
+  fput "stow #{GameObj.left_hand.noun}" if GameObj.left_hand.name != 'Empty'
+  case DRC.bput('get prayer chain', 'You get', 'I could not find', 'What were you referring to', 'already in your inventory')
+  when 'I could not find', 'What were you referring to'
+    echo 'No Prayer Chain Found - Skipping Bead Carve'
+    return
+  when 'already in your inventory'
+    fput 'remove my prayer chain'
+  end
+  beads = DRC.bput("look at my chain", /^Strung on to the prayer bead chain you see .*\./, 'There are currently no beads on it.')
+  bead_count = DRC.list_to_nouns(beads.match(/^Strung on to the prayer bead chain you see (.*)\./).to_a[1])
+  return if bead_count.count >= 10 
+  put_in_container('prayer chain')
+  DRC.forage? 'limb'
+  get_from_container($settings.water_holder)
```

Comment:
You might consider checking for a carving knife, shaper, etc. before starting the process. `common-items` offers an `exists?` method that might help.